components/masking_functions/src/masking_functions/registration_routines.cpp

```diff
@@ -0,0 +1,1262 @@
+      sresult = masking_functions::mask_inner(sresult, 0, 8, masking_char);
+      return masking_functions::mask_inner(sresult, 8, 0, masking_char);
+    } else {
+      // otherwise (no delimiters at all, or just one at an unknown position),
+      // we use 'mask_inner_alphanum()' for the whole range
+      return masking_functions::mask_inner_alphanum(cs_str, 0, 0, masking_char);
+    }
+  }
+};
+
+//
+// mask_iban(string, [char])
+//
+// IBAN consists of 2 upper-case latin letters (country code) followed by
+// 13..32 latin alphanumeric characters [A-Z0-9]. It may include up to 8
+// delimiters
+// E.g. ZZ0123456789012 or IE12 BOFI 9000 0112 3456 78
+//      ZZ************* or IE** **** **** **** **** **
+// min_length          : 15
+// max_length          : 42
+// default_masking_char: *
+class mask_iban_impl final : private mask_impl_base {
+ public:
+  using mask_impl_base::calculate;
+  using mask_impl_base::mask_impl_base;
+
+ private:
+  virtual std::size_t min_length() const override { return 15; }
+  virtual std::size_t max_length() const override { return 34 + 8; }
+  virtual std::string_view default_ascii_masking_char() const override {
+    return star_ascii_masking_char;
+  }
+  virtual masking_functions::charset_string process(
+      const masking_functions::charset_string &cs_str,
+      const masking_functions::charset_string &masking_char) const override {
+    // as positions of the delimiters may vary, we always use the
+    // 'mask_inner_alphanum()' function for everything except for the first 2
+    // characters
+    return masking_functions::mask_inner_alphanum(cs_str, 2, 0, masking_char);
+  }
+};
+
+//
+// mask_pan(string, [char])
+//
+// Card number consists of 14..16 digits with up to 3 delimiters
+// separated with a delimiter
+// E.g. 1234567887654321 or 1234 5678 8765 4321
+//      XXXXXXXXXXXX4321 or XXXX XXXX XXXX 4321
+// min_length          : 14
+// max_length          : 19
+// default_masking_char: X
+class mask_pan_impl final : private mask_impl_base {
+ public:
+  using mask_impl_base::calculate;
+  using mask_impl_base::mask_impl_base;
+
+ private:
+  virtual std::size_t min_length() const override { return 14; }
+  virtual std::size_t max_length() const override { return 19; }
+  virtual std::string_view default_ascii_masking_char() const override {
+    return x_ascii_masking_char;
+  }
+
+  virtual masking_functions::charset_string process(
+      const masking_functions::charset_string &cs_str,
+      const masking_functions::charset_string &masking_char) const override {
+    // as positions of the delimiters may vary, we always use the
+    // 'mask_inner_alphanum()' function for everything except for the last 4
+    // characters
+    return masking_functions::mask_inner_alphanum(cs_str, 0, 4, masking_char);
+  }
+};
+
+//
+// mask_pan_relaxed(string, [char])
+//
+// Card number consists of 14..16 digits with up to 3 delimiters
```

Comment:
I think "with up to 3 delimiters separated with a delimiter" should be replaced with "possible separated with up to 3 delimiters". What do you think? Same applies to mask_pan description.

Reply:
fixed